orttraining/orttraining/test/python/orttraining_ortmodule_tests.py

```diff
@@ -26,10 +26,29 @@ def parse_arguments():
 def run_ortmodule_api_tests(cwd, log):
     log.debug('Running: ORTModule-API tests')
 
-    command = [sys.executable, '-m', 'pytest', '-sv', 'orttraining_test_ortmodule_api.py']
-
-    run_subprocess(command, cwd=cwd, log=log).check_returncode()
-
+    class TestNameCollecterPlugin:
+        def __init__(self):
+            self.collected = set()
+
+        def pytest_collection_modifyitems(self, items):
+            for item in items:
+                print('item.name: ', item.name)
+                self.collected.add(item.name)
+    
+    import os
+    import pytest
+    plugin = TestNameCollecterPlugin()
+    print(cwd)
+    test_script_filename = os.path.join("orttraining_test_ortmodule_api.py")
+    pytest.main(['--collect-only', test_script_filename], plugins=[plugin])
+
+    # TODO: FIX THIS! 
+    # Running tests in a loop one after another, 
+    # because ORTModule doesn't support multiple run call at the same time
+    for test_name in plugin.collected:
+        run_subprocess([
+            sys.executable, '-m', 'pytest',
+            'orttraining_test_ortmodule_api.py', '-sv', '-k', test_name], cwd=cwd).check_returncode()
```

Comment:
It seems this refactoring hides an important limitation on the new implementation, which is support for mode than one ORTModule in memory

It is better to have a broken tested disabled than having it changed so that it doesnt guard the system agains an intended feature

Reply:
There are a lot of unit test uses pytest.mark.parametrize(), they runs multiple test cases can destroy them in the end.

we will basically need to disable all the tests if we keep running the unit test in the old way. 
This is tracked in ADO: https://msdata.visualstudio.com/Vienna/_sprints/taskboard/ONNX%20Training/Vienna/Cobalt?workitem=1039183

Reply:
Seems we use similar things for frontend test:
https://github.com/microsoft/onnxruntime/blob/13d7db9a98542eb40399592684309ff7dd259551/tools/ci_build/build.py#L1230-L1237

Reply:
ORTTrainer uses this for a different (known) bug (on TrainingSession). ORTModule use InferenceSession instead